services/blockchain-app-registry/shared/metadata.js

```diff
@@ -330,8 +344,131 @@ const getBlockchainAppsTokenMetadata = async (params) => {
 	return blockchainAppsTokenMetadata;
 };
 
+const resolveTokenMetaInfo = async (tokenInfoFromDB) => {
+	const tokensMeta = [];
+
+	await BluebirdPromise.map(
+		tokenInfoFromDB,
+		async (entry) => {
+			const parsedTokenMeta = await readMetadataFromClonedRepo(
+				entry.network,
+				entry.chainName,
+				config.FILENAME.NATIVETOKENS_JSON,
+			);
+
+			parsedTokenMeta.tokens.forEach(async token => {
+				tokensMeta.push({
+					...token,
+					chainID: entry.chainID,
+					chainName: entry.chainName,
+					network: entry.network,
+				});
+			});
+		},
+		{ concurrency: tokenInfoFromDB.length },
+	);
+
+	return tokensMeta;
+};
+
+const getSupportedTokensFromServiceURLs = async (serviceURLs) => {
+	for (let i = 0; i < serviceURLs.length; i++) {
+		const tokenSummaryEndpoint = `${serviceURLs[i].http}/api/v3/token/summary`;
+		// eslint-disable-next-line no-await-in-loop
+		const { data: response } = await HTTP.request(tokenSummaryEndpoint);
+
+		if (response.data && response.data.supportedTokens) {
+			return Promise.resolve(response.data.supportedTokens);
+		}
+	}
+	return Promise.resolve({});
+};
+
+const getAllTokensMetaInNetworkByChainID = async (chainID, limit, offset, sort) => {
+	const tokenMetadataTable = await getTokenMetadataIndex();
+	const searchParams = {
+		search: {
+			property: 'tokenID',
+			startsWith: chainID.substring(0, 2),
+		},
+		limit,
+		offset,
+		sort,
+	};
+	const tokensResultSet = await tokenMetadataTable.find(searchParams, ['network', 'chainID', 'chainName']);
+	const total = await tokenMetadataTable.count(searchParams, ['network', 'chainID', 'chainName']);
+	const tokensMeta = await resolveTokenMetaInfo(tokensResultSet);
+	// Fetch the data
+	return { tokensMeta, total };
+};
+
+const getTokensMetaByTokenIDs = async (patternTokenIDs, exactTokenIDs, limit, offset, sort) => {
+	const tokenMetadataTable = await getTokenMetadataIndex();
+	const searchParams = {
+		whereIn: [{
+			property: 'tokenID',
+			values: exactTokenIDs,
+		}],
+		orSearch: patternTokenIDs.map(e => {
+			const chainID = e.substring(0, LENGTH_CHAIN_ID);
+			return {
+				property: 'tokenID',
+				startsWith: chainID,
+			};
+		}),
+		limit,
+		offset,
+		sort,
+	};
+
+	const tokensResultSet = await tokenMetadataTable.find(searchParams, ['network', 'chainID', 'chainName']);
+	const total = await tokenMetadataTable.count(searchParams, ['network', 'chainID', 'chainName']);
```

Comment:
No, it is necessary to make the second call, since `tokensResultSet` will respect the pagination params and will return a resultSet of size that is less than or equal to `searchParams.limit`.